controllers/storageclusterpeer/storageclusterpeer_controller.go

```diff
@@ -18,45 +18,311 @@ package storageclusterpeer
+		Watches(&rookCephv1.CephBlockPool{}, enqueueStorageClusterPeerRequest).
 		Complete(r)
 }
+
+func (r *StorageClusterPeerReconciler) reconcilePhases() (reconcile.Result, error) {
+	r.Log.Info("Running StorageClusterPeer controller")
+
+	//marked for deletion
+	if !r.storageClusterPeer.GetDeletionTimestamp().IsZero() {
+		r.storageClusterPeer.Status.Phase = ocsv1.StorageClusterPeerDeleting
+
+		ocsClient, err := r.newExternalClient()
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		defer ocsClient.Close()
+
+		if err := r.reconcileRevokeBlockPoolPeering(ocsClient); err != nil {
+			return reconcile.Result{}, err
+		}
+
+		//remove finalizer
+		r.Log.Info("removing finalizer from the StorageClusterPeer resource")
+		r.storageClusterPeer.SetFinalizers(slices.Filter([]string{}, r.storageClusterPeer.GetFinalizers(), func(s string) bool {
+			return s != ocsv1.StorageClusterPeerFinalizer
+		}))
+		if err := r.Client.Update(r.ctx, r.storageClusterPeer); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to remove finalizer from StorageClusterPeer: %v", err)
+		}
+		r.Log.Info("finalizer removed successfully")
+		return reconcile.Result{}, nil
+	}
+	//not marked for deletion
+
+	// add finalizer
+	slices.Contains(r.storageClusterPeer.GetFinalizers(), ocsv1.StorageClusterPeerFinalizer)
+	if !slices.Contains(r.storageClusterPeer.GetFinalizers(), ocsv1.StorageClusterPeerFinalizer) {
+		r.Log.V(-1).Info("finalizer missing on the StorageClusterPeer resource, adding...")
+		r.storageClusterPeer.SetFinalizers(append(r.storageClusterPeer.GetFinalizers(), ocsv1.StorageClusterPeerFinalizer))
+		if err := r.update(r.storageClusterPeer); err != nil {
+			return reconcile.Result{}, fmt.Errorf("failed to update StorageClusterPeer with finalizer: %v", err)
+		}
+	}
+
+	ocsClient, err := r.newExternalClient()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	defer ocsClient.Close()
+
+	r.storageClusterPeer.Status.Phase = ocsv1.StorageClusterPeerConfiguring
+
+	if err := r.reconcileCephBlockPool(); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if err := r.reconcilePeerBlockPool(ocsClient); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	r.storageClusterPeer.Status.Phase = ocsv1.StorageClusterPeerReady
+
+	return reconcile.Result{}, nil
+}
+
+func (r *StorageClusterPeerReconciler) newExternalClient() (*providerClient.OCSProviderClient, error) {
+	ocsClient, err := providerClient.NewProviderClient(r.ctx, r.storageClusterPeer.Spec.OCSAPIServerURI, time.Second*10)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create a new ocs client: %v", err)
+	}
+	return ocsClient, nil
+}
+
+func (r *StorageClusterPeerReconciler) reconcileCephBlockPool() error {
+
+	for _, cephBlockPool := range r.cephBlockPoolList.Items {
+
+		_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, &cephBlockPool, func() error {
+			cephBlockPool.Spec.Mirroring.Enabled = true
+			cephBlockPool.Spec.Mirroring.Mode = "image"
+			return nil
+		})
```

Comment:
is there any reason in enabling mirroring daemon via gRPC but not locally?

Reply:
The mirror daemon is required on the secondary Cluster not on primary cluster